src/NuGet.Clients/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -51,6 +51,11 @@ public class PushCommand : Command
                     NoSymbols,
                     Console);
             }
+            catch (TaskCanceledException ex)
+            {
+                string timeoutMessage = LocalizedResourceManager.GetString(nameof(NuGetResources.PushCommandTimeoutError));
+                throw new AggregateException(ex, new Exception(timeoutMessage));
```

Comment:
I would expect this to throw a more specific exception type. This works because it is just caught and written out to the console?